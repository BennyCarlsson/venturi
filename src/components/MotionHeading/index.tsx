import Heading, { HeadingProps } from 'components/Heading'
import { HTMLMotionProps, motion } from 'framer-motion'

interface MotionHeadingProps extends HTMLMotionProps<'h1'>, HeadingProps {}

const MotionHeading = (_props: MotionHeadingProps) => {
  const getAs = () => {
    switch (_props.headingType) {
      case 'h1':
        return motion.h1
      case 'h2':
        return motion.h2
      case 'h3':
        return motion.h3
      case 'h4':
        return motion.h4
      case 'h5':
        return motion.h5
      case 'h6':
        return motion.h6
      default:
        return motion.h2
    }
  }

  const props = { as: getAs(), ..._props }
  return <Heading {...props} />
}

export default MotionHeading
