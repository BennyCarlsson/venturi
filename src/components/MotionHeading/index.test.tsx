import { render, screen } from '@testing-library/react'
import renderer from 'react-test-renderer'
import MotionHeading from '.'

describe('<MotionHeading />', () => {
  it('should match snapshot', () => {
    const tree = renderer
      .create(<MotionHeading headingType={'h2'}>Heading</MotionHeading>)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should render without crashing', () => {
    render(<MotionHeading headingType={'h2'}>Heading</MotionHeading>)
    expect(screen).toBeTruthy()
  })

  it('should render text', () => {
    render(<MotionHeading headingType={'h2'}>Heading</MotionHeading>)
    expect(screen.getByText('Heading')).toBeInTheDocument()
  })

  type headingType = 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6'
  const headingTypes: headingType[] = ['h1', 'h2', 'h3', 'h4', 'h5', 'h6']
  headingTypes.forEach((headingType) => {
    it(`${headingType} should render passed child element`, () => {
      const text = 'Testy test'
      render(<MotionHeading headingType={headingType}>{text}</MotionHeading>)
      expect(
        screen.getByText(text, { selector: headingType })
      ).toBeInTheDocument()
    })
  })
})
