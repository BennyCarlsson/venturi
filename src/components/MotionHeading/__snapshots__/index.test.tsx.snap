// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MotionHeading /> should match snapshot 1`] = `
<h2
  className="sc-gsTCUz clJLCV"
  style={Object {}}
>
  Heading
</h2>
`;
